shell/platform/darwin/macos/framework/Source/FlutterViewController.mm

```diff
@@ -500,6 +505,12 @@ - (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
   }
 }
 
+- (void)playSystemSound:(NSString*)soundType {
+  if ([soundType isEqualToString:@"SystemSoundType.alert"]) {
+    AudioServicesPlayAlertSound(kSystemSoundID_UserPreferredAlert);
```

Comment:
The header says `AudioServicesPlayAlertSound` will be deprecated in a future release and replaced by `AudioServicesPlayAlertSoundWithCompletion`.  Maybe pass in a completion block that calls `result(nil)`?
```objective-c
  [self playSystemSound:call.arguments completion:^{
    result(nil);
  }];
```
```objective-c
- (void)playSystemSound:(NSString*)soundType completion:(void (^)(void))completion {
  if ([soundType isEqualToString:@"SystemSoundType.alert"]) {
    AudioServicesPlayAlertSoundWithCompletion(kSystemSoundID_UserPreferredAlert, completion);
  }
}
```

Reply:
Why not just call `NSBeep()`?